HeterogeneousCore/CUDAUtilities/interface/cudastdAlgorithm.h

```diff
@@ -0,0 +1,70 @@
+#ifndef HeterogeneousCore_CUDAUtilities_cudastdAlgorithm_h
+#define HeterogeneousCore_CUDAUtilities_cudastdAlgorithm_h
+
+#include <utility>
+
+#include <cuda_runtime.h>
+
+// reimplementation of std algorithms able to compile with CUDA and run on GPUs,
+// mostly by declaringthem constexpr
+
+namespace cuda_std {
```

Comment:
it looks like all other "namespace cuda" is inside cms::.
Is this one intentional or was just missed?

Reply:
I don't know if it was intentional or not, but I think the distinction should be kept.

Here we add support for C++ std functions and classes that are not (yet) available in CUDA.

We can likely transition away from then when
  - CUDA supports c++17
  - libcu++ gets some decent documentation